wireapp/wire-android
# feat: audio message filter (WPB-6406)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6406" title="WPB-6406" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-6406</a>  [Android] Apply Filters to Audio Recording
  </td></table>
  <br />
 






----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

There was no way into adding filters to the recorded audio message.

### Causes (Optional)

It was not implemented.

### Solutions

Add a checkbox on audio component (before recording) to add audio effects to the recorded audio.
Audio with effects is also playable at the last step of audio component.

### Dependencies (Optional)

Needs releases with:

- [x] https://github.com/wireapp/kalium/pull/2686

### Testing

#### Test Coverage (Optional)


#### How to Test

- Open Audio component
- Check the checkbox to apply filter
- start recording, talk a bit, stop recording
- in ready to send component, playable audio is the one with effects
- send audio
- sent audio is the one with effects and is playable in all platforms

### Attachments (Optional)
<img src="https://github.com/wireapp/wire-android/assets/5890660/79576e83-efb8-4df8-b507-6f6d2aad9c9d" width="200" height="400" alt="ready to record audio with effects" />
<img src="https://github.com/wireapp/wire-android/assets/5890660/37e4589b-3102-42af-ab8f-e535b894c8c8" width="200" height="400" alt="recording audio with effects" />
<img src="https://github.com/wireapp/wire-android/assets/5890660/4885f57e-22c4-4ba7-8153-3eb5bcfbd964" width="200" height="400" alt="ready to send audio with effects" />
<img src="https://github.com/wireapp/wire-android/assets/5890660/3e6b0b57-49eb-4be4-bfe8-4d7e1b291672" width="200" height="400" alt="audio with effects sent" />
